src/core/unit.c

```diff
@@ -5267,6 +5289,59 @@ static int unit_export_log_extra_fields(Unit *u, const ExecContext *c) {
         return r;
 }
 
+static int unit_export_log_rate_limit_interval(Unit *u, const ExecContext *c) {
+        char buf[FORMAT_TIMESPAN_MAX];
+        const char *p;
+        int r;
+
+        assert(u);
+        assert(c);
+
+        if (u->exported_log_rate_limit_interval)
+                return 0;
+
+        if (c->log_rate_limit_interval_usec == 0)
+                return 0;
+
+        p = strjoina("/run/systemd/units/log-rate-limit-interval:", u->id);
+        format_timespan(buf, sizeof(buf), c->log_rate_limit_interval_usec, USEC_PER_SEC);
+
+        r = symlink_atomic(buf, p);
+        if (r < 0)
+                return log_unit_debug_errno(u, r, "Failed to create log rate limit interval symlink %s: %m", p);
+
+        u->exported_log_rate_limit_interval = true;
+        return 0;
+}
+
+static int unit_export_log_rate_limit_burst(Unit *u, const ExecContext *c) {
+        _cleanup_free_ char *buf = NULL;
+        const char *p;
+        int r;
+
+        assert(u);
+        assert(c);
+
+        if (u->exported_log_rate_limit_burst)
+                return 0;
+
+        if (c->log_rate_limit_burst == 0)
+                return 0;
+
+        p = strjoina("/run/systemd/units/log-rate-limit-burst:", u->id);
+
+        r = asprintf(&buf, "%u", c->log_rate_limit_burst);
+        if (r < 0)
+                return log_unit_debug_errno(u, r, "Failed to write log rate limit burst to buffer: %m");
```

Comment:
`asprintf()` returns -1 on error, and stores the error number in `errno`, hence it isn't right to pass `r` to log_unit_debug_errno() as error number here...

Moreover, the only case where asprintf() fails is out of memory, and those conditions we usually handle with "log_oom()"...

i.e. consider rewriting as:

```c
if (asprintf(…) < 0)
        return log_oom();
```